In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import ttest_ind

import warnings
warnings.filterwarnings("ignore")

# 데이터 불러오기
picher_file_path = '../data/picher_stats_2017.csv'
batter_file_path = '../data/batter_stats_2017.csv'
picher = pd.read_csv(picher_file_path)
batter = pd.read_csv(batter_file_path)

# 팀 정보, 연봉 정보 포함된 데이터프레임 생성
baseball = pd.concat([picher, batter])



# 한화와 SK 데이터 추출
HanWha = baseball.loc[baseball['팀명'] == '한화']
SK = baseball.loc[baseball['팀명'] == 'SK']

# 각 팀의 연봉 평균 계산
HanWha_salary = HanWha['연봉(2017)'].mean()
SK_salary = SK['연봉(2017)'].mean()

# t-test로 두 팀 간의 차이 검정
t_stat, p_val = ttest_ind(HanWha['연봉(2017)'], SK['연봉(2017)'], equal_var=False)

print('한화 평균 연봉:', HanWha_salary)
print('SK 평균 연봉:', SK_salary)
print('t-test 통계량:', t_stat)
print('p-value:', p_val)

한화 평균 연봉: 22253.846153846152
SK 평균 연봉: 17736.842105263157
t-test 통계량: 0.6624041217840793
p-value: 0.5099280465043418





 'picher_stats_2017.csv'와 'batter_stats_2017.csv' 파일을 불러와 picher와 batter 변수에 저장
 각각 투수와 타자의 선수명, 팀명, 나이, 방어율, 타율 등의 정보 포함됨.

 concat 함수를 이용하여 picher와 batter 데이터프레임을 합쳐 baseball 변수에 저장
 
 loc 함수를 이용하여 한화와 SK의 데이터 추출하여 HanWha와 SK 변수에 저장

 HanWha와 SK 팀의 평균 연봉을 계산하여 각각 HanWha_salary와 SK_salary 변수에 저장

 t-test를 이용하여 두 팀 간의 연봉 차이를 검정
 ttest_ind 함수를 이용하고, equal_var 인자를 False로 지정하여 등분산 가정을 하지 않도록 함
 검정 결과인 t-test 통계량과 p-value 값을 각각 t_stat과 p_val 변수에 저장

 '한화 평균 연봉:', 'SK 평균 연봉:', 't-test 통계량:', 'p-value:' 라는 문구와 함께 각 값이 출력


해당 t-test 결과에서 p-value가 0.5099로 나왔으므로, 유의수준 0.05에서 검정했을 때 p_value가 높기 때문에 
한화와 SK의 연봉 평균 간에는 통계적으로 유의미한 차이가 없다고 할 수 있다.